ProcessMembership/handler.ts

```diff
@@ -3,70 +3,188 @@ import { flow, pipe } from "fp-ts/lib/function";
+  // respectively for surname and name
+  else {
+    const name = CODICEFISCALE.substring(3, 6);
+    const surname = CODICEFISCALE.substring(0, 3);
+    return { name, surname };
+  }
+};
+const toSelfcareRole = (role: PecDelegate["TIPODELEGATO"]): UserDto["role"] => {
+  if (role === "Principale") {
+    return RoleEnum.MANAGER;
+  }
+  // Any other parsed role, we assign the DELEGATE role
+  else {
+    return RoleEnum.DELEGATE;
+  }
+};
+
+const composeSelfcareUser = (input: PecDelegate): UserDto => ({
+  email: input.EMAIL,
+  role: toSelfcareRole(input.TIPODELEGATO),
+  taxCode: input.CODICEFISCALE,
+  ...formatDelegateNameAndSurname(input)
+});
+
+const composeSelfcareContract = (input: IContract): ImportContractDto => ({
+  contractType: input.version,
+  fileName: input.attachment.name,
+  filePath: input.attachment.path
+});
 
 // Prepare data to be sent to SelfCare
 const composeSelfCareMembershipClaim = (
-  _ipaCode: IpaCode,
-  _contract: IContract
-): SelfCareMembershipClaimParams => ({});
+  fiscalCode: NonEmptyString,
+  contract: IContractWithDelegates
+): SelfCareMembershipClaimParams => ({
+  body: {
+    importContract: composeSelfcareContract(contract),
+    users: contract.delegates.map(composeSelfcareUser)
+  },
+  externalInstitutionId: fiscalCode
+});
 
 // Submit the claim to SelfCare to import the memebership
-const submitMembershipClaimToSelfcare = (
-  _selfcareConfig: unknown /* TBD */
-) => (_claim: SelfCareMembershipClaimParams): TE.TaskEither<Error, void> =>
-  TE.left(
-    new NotImplementedError(
-      "submitMembershipClaimToSelfcare() - Not implemented yet"
-    )
+const submitMembershipClaimToSelfcare = (selfcareClient: SelfCareClient) => (
+  claim: SelfCareMembershipClaimParams
+): TE.TaskEither<Error, void> =>
+  pipe(
+    TE.tryCatch(
+      () => selfcareClient.contractOnboardingUsingPOST(claim),
+      E.toError
+    ),
+    TE.map(_ => void 0)
   );
 
 // Save that the memebeship is not meant to be processed by the current business logic
-const markMembershipAsDiscarded = (_dao: Dao) => (
-  _ipaCode: IpaCode
-): TE.TaskEither<Error, ReadonlyArray<IContract>> =>
-  TE.left(new NotImplementedError("discardMembership() - Not implemented yet"));
+const markMembership = (dao: Dao) => (
+  ipaCode: IpaCode,
+  status: MembershipStatus
+): TE.TaskEither<Error, void> =>
+  pipe(
+    TE.tryCatch(() => dao("memberships").readItemById(ipaCode), E.toError),
+    TE.chain(
+      flow(
+        IMembership.decode,
+        TE.fromEither,
+        TE.mapLeft(flow(readableReport, E.toError))
+      )
+    ),
+    TE.chain(m =>
+      TE.tryCatch(() => dao("memberships").upsert({ ...m, status }), E.toError)
+    ),
+    TE.map(_ => void 0)
+  );
+
+// Save that the memebeship is not meant to be processed by the current business logic
+const markMembershipAsDiscarded = (dao: Dao) => (
+  ipaCode: IpaCode
+): TE.TaskEither<Error, void> => markMembership(dao)(ipaCode, "Discarded");
```

Comment:
I think it will be useful to store the reason of the discarded item (for example setting `note` field)